nav2_graceful_controller/src/smooth_control_law.cpp

```diff
@@ -59,13 +58,15 @@ geometry_msgs::msg::Twist SmoothControlLaw::calculateRegularVelocity(
   auto ego_coords = EgocentricPolarCoordinates(target, current, backward);
   // Calculate the curvature
   double curvature = calculateCurvature(ego_coords.r, ego_coords.phi, ego_coords.delta);
+  // Invert the curvature if the robot is moving backwards
+  curvature = backward ? -curvature : curvature;
 
   // Adjust the linear velocity as a function of the path curvature to
   // slowdown the controller as it approaches its target
   double v = v_linear_max_ / (1.0 + beta_ * std::pow(fabs(curvature), lambda_));
 
   // Slowdown when the robot is near the target to remove singularity
-  v = std::min(v_linear_max_ * (ego_coords.r / slowdown_radius_), v);
+  v = std::min(v_linear_max_ * fabs(ego_coords.r / slowdown_radius_), v);
```

Comment:
isn't `r` always positive now? Shouldn't `(ego_coords.r / slowdown_radius_)` always be positive?

Reply:
yeap, that is not necessary anymore.